### PR TITLE
[major_changes] Added Cloud Account feature for cloud APICs.

### PR DESCRIPTION
**Pull request by anvitha-jain**
solves #545 #676

### CODE REVIEW

File: aci/resource_aci_cloudaccount.go [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+		_, newRelParam := d.GetChange("relation_cloud_rs_account_to_access_policy")
+		err = aciClient.DeleteRelationcloudRsAccountToAccessPolicy(cloudAccount.DistinguishedName)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		err = aciClient.CreateRelationcloudRsAccountToAccessPolicy(cloudAccount.DistinguishedName, cloudAccountAttr.Annotation, newRelParam.(string))
+
+		if err != nil {
+			return diag.FromErr(err)
+		}
```
Comment (lhercot) [inline aci/resource_aci_cloudaccount.go:287]: shouldn't there be a condition on the recreation? What happens if the relationship is not defined anymore?

File: aci/resource_aci_cloudaccount.go [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+
+	}
+
+	d.SetId(cloudAccount.DistinguishedName)
+	log.Printf("[DEBUG] %s: Update finished successfully", d.Id())
+	return resourceAciAccountRead(ctx, d, m)
+}
+
+func resourceAciAccountRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
```
Comment (lhercot) [inline aci/resource_aci_cloudaccount.go:309]: resourceAciAccountRead -> resourceAciCloudAccountRead

File: aci/resource_aci_cloudaccount.go [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+	log.Printf("[DEBUG] %s: Beginning Read", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+
+	cloudAccount, err := getRemoteAccount(aciClient, dn)
+	if err != nil {
+		// d.SetId("")
```
Comment (lhercot) [inline aci/resource_aci_cloudaccount.go:316]: do we need this line?

File: aci/resource_aci_cloudaccount.go [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+		return diag.FromErr(err)
+	}
+
+	_, err = setAccountAttributes(cloudAccount, d)
+	if err != nil {
+		// d.SetId("")
```
Comment (lhercot) [inline aci/resource_aci_cloudaccount.go:322]: do we need this line?

File: aci/resource_aci_cloudaccount.go [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+	cloudRsCredentialsData, err := aciClient.ReadRelationcloudRsCredentials(dn)
+	if err != nil {
+		log.Printf("[DEBUG] Error while reading relation cloudRsCredentials %v", err)
+		d.Set("relation_cloud_rs_credentials", "")
+	} else {
+		if _, ok := d.GetOk("relation_cloud_rs_credentials"); ok {
+			tfName := d.Get("relation_cloud_rs_credentials").(string)
+			if tfName != cloudRsCredentialsData {
+				d.Set("relation_cloud_rs_credentials", "")
+			}
```
Comment (lhercot) [inline aci/resource_aci_cloudaccount.go:348]: Remove this part

File: examples/cloud_azure_account/main.tf [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    aci = {
+      source = "ciscodevnet/aci"
+    }
+  }
+}
+
+provider "aci" {
+  username = "ansible_github_ci"
+  password = "sJ94G92#8dq2hx*K4qh"
+  url      = "https://104.42.26.226/"
```
Comment (lhercot) [inline examples/cloud_azure_account/main.tf:12]: Clean this

File: examples/cloud_azure_account/main.tf [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  insecure = true
+}
+
+resource "aci_tenant" "footenant" {
+  description = "sample aci_tenant from terraform"
+  name        = "test_tenant"
+  annotation  = "tag_tenant"
+  name_alias  = "alias_tenant"
```
Comment (lhercot) [inline examples/cloud_azure_account/main.tf:20]: don't need those in this example

File: examples/cloud_azure_account/main.tf [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+resource "aci_cloud_azure_account" "azure-account" {
+  tenant_dn         = aci_tenant.footenant.id
+  # description       = "zure account config"
+  access_type       = "managed"  //credentials or managed
+  account_id        = "test123"
```
Comment (lhercot) [inline examples/cloud_azure_account/main.tf:27]: use somehting like my_example_account_id or something

File: examples/cloud_azure_account/main.tf [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+resource "aci_tenant" "footenant" {
+  description = "sample aci_tenant from terraform"
+  name        = "test_tenant"
+  annotation  = "tag_tenant"
+  name_alias  = "alias_tenant"
+}
+
+resource "aci_cloud_azure_account" "azure-account" {
+  tenant_dn         = aci_tenant.footenant.id
+  # description       = "zure account config"
```
Comment (lhercot) [inline examples/cloud_azure_account/main.tf:25]: zure?

File: website/docs/d/cloud_azure_account.html.markdown [hunk 10]
```diff
@@ -0,0 +1,47 @@
+---
+layout: "aci"
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:2]: add subcategory

File: website/docs/d/cloud_azure_account.html.markdown [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+page_title: "ACI: aci_account"
+sidebar_current: "docs-aci-data-source-account"
+description: |-
+  Data source for ACI Account
+---
+
+# aci_account #
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:9]: aci_cloud_account

File: website/docs/d/cloud_azure_account.html.markdown [hunk 12]
```diff
@@ -0,0 +1,47 @@
+---
+layout: "aci"
+page_title: "ACI: aci_account"
+sidebar_current: "docs-aci-data-source-account"
+description: |-
+  Data source for ACI Account
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:6]: ACI Cloud Account

File: website/docs/d/cloud_azure_account.html.markdown [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+---
+
+# aci_account #
+
+Data source for ACI Account
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:11]: ACI Cloud Account

File: website/docs/d/cloud_azure_account.html.markdown [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+
+
+## API Information ##
+
+* `Class` - cloudAccount
+* `Distinguished Name` - uni/tn-{name}/act-[{id}]-vendor-{vendor}
+
+## GUI Information ##
+
+* `Location` -
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:21]: Add UI location

File: website/docs/d/cloud_azure_account.html.markdown [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+data "aci_account" "example" {
+  tenant_dn  = aci_tenant.example.id
+  account_id  = "example"
+  vendor  = "example"
+}
+```
+
+## Argument Reference ##
+
+* `tenant_dn` - (Required) Distinguished name of parent Tenant object.
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:37]: of the parent Tenant object

File: website/docs/d/cloud_azure_account.html.markdown [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+* `account_id` - (Required) Account_id of object Account.
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:38]: ID of the Cloud Account object

File: website/docs/d/cloud_azure_account.html.markdown [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+* `vendor` - (Required) Vendor of object Account.
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:39]: Vendor of the Cloud Account object

File: website/docs/d/cloud_azure_account.html.markdown [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the Account.
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:42]: of the Cloud Account

File: website/docs/d/cloud_azure_account.html.markdown [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+* `annotation` - (Optional) Annotation of object Account.
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:43]: of the Cloud Account object

File: website/docs/d/cloud_azure_account.html.markdown [hunk 20]
```diff
@@ -0,0 +1,47 @@
...
+* `name_alias` - (Optional) Name Alias of object Account.
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:44]: of the Cloud Account object

File: website/docs/d/cloud_azure_account.html.markdown [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+* `access_type` - (Optional) Account acccess type. How to authenticate to the account (managed=no credentials required (IAM), credentials=using accessKeys)
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:45]: I am not a fan of the second part can you rephrase it.

File: website/docs/d/cloud_azure_account.html.markdown [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+* `account_id` - (Optional) Account id. An object identifier.
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:46]: remove it from here as part of required attribute

File: website/docs/d/cloud_azure_account.html.markdown [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+* `vendor` - (Optional) Account vendor. The vendor of the controller.
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:47]: remove it from here as part of required attribute

File: website/docs/d/cloud_azure_account.html.markdown [hunk 24]
```diff
@@ -0,0 +1,47 @@
...
+
+# aci_account #
+
+Data source for ACI Account
+
+
+## API Information ##
+
+* `Class` - cloudAccount
+* `Distinguished Name` - uni/tn-{name}/act-[{id}]-vendor-{vendor}
```
Comment (lhercot) [inline website/docs/d/cloud_azure_account.html.markdown:17]: name -> tenant_name
id -> account_id

File: website/docs/r/cloud_azure_account.html.markdown [hunk 25]
```diff
@@ -0,0 +1,65 @@
+---
```
Comment (lhercot) [inline website/docs/r/cloud_azure_account.html.markdown:1]: Apply same comments as from data source

**Review by lhercot — CHANGES_REQUESTED** [review]

File: aci/resource_aci_cloudaccount.go [hunk 26]
```diff
@@ -0,0 +1,368 @@
...
+				Required: true,
+				ForceNew: true,
+			},
+			"access_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"credentials",
+					"managed",
```
[the remark this answers is not in the document]
Comment (anvitha-jain) [reply aci/resource_aci_cloudaccount.go:39]: It is only 2 in azure

File: aci/resource_aci_fvrscloudaccount.go [hunk 27]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+func getRemoteTenanttoCloudAccountAssociation(client *client.Client, dn string) (*models.TenanttoCloudAccountAssociation, error) {
+	fvRsCloudAccountCont, err := client.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+	fvRsCloudAccount := models.TenanttoCloudAccountAssociationFromContainer(fvRsCloudAccountCont)
+	if fvRsCloudAccount.DistinguishedName == "" {
+		return nil, fmt.Errorf("TenanttoCloudAccountAssociation %s not found", fvRsCloudAccount.DistinguishedName)
```
Comment (lhercot) [inline aci/resource_aci_fvrscloudaccount.go:48]: Tenant to Cloud Account Association %s

File: examples/aci_cloud_account/main.tf [hunk 28]
```diff
@@ -0,0 +1,49 @@
+terraform {
```
Comment (lhercot) [inline examples/aci_cloud_account/main.tf:1]: Merge all into a single example

File: website/docs/d/cloud_account.html.markdown [hunk 29]
```diff
@@ -0,0 +1,47 @@
...
+
+
+## API Information ##
+
+* `Class` - cloudAccount
+* `Distinguished Name` - uni/tn-{tenant_name}/act-[{account_id}]-vendor-{vendor}
+
+## GUI Information ##
+
+* `Location` - Cloud APIC -> Application Management -> Tenants
```
Comment (lhercot) [inline website/docs/d/cloud_account.html.markdown:22]: Cloud APIC -> Application Management -> Tenants -> {tenant_name}

File: website/docs/d/cloud_account.html.markdown [hunk 30]
```diff
@@ -0,0 +1,47 @@
...
+* `tenant_dn` - (Required) Distinguished name of the parent Tenant object.
+* `account_id` - (Required) ID of the Cloud Account object.
+* `vendor` - (Required) Vendor of the Cloud Account object.
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the Cloud Account.
+* `name` - (Optional) Name of the Cloud Account object.
+* `annotation` - (Optional) Annotation of the Cloud Account object.
+* `name_alias` - (Optional) Name Alias of the Cloud Account object.
+* `access_type` - (Optional) Authentication to the Cloud Account (managed=no credentials required (IAM), credentials=using accessKeys).
```
Comment (lhercot) [inline website/docs/d/cloud_account.html.markdown:47]: Authentication type for the Cloud Account

File: website/docs/d/tenant_to_cloud_account.html.markdown [hunk 31]
```diff
@@ -0,0 +1,44 @@
...
+
+
+## API Information ##
+
+* `Class` - fvRsCloudAccount
+* `Distinguished Name` - uni/tn-{tenant_name}/rsCloudAccount
+
+## GUI Information ##
+
+* `Location` - Cloud APIC -> Application Management -> Tenants
```
Comment (lhercot) [inline website/docs/d/tenant_to_cloud_account.html.markdown:22]: Cloud APIC -> Application Management -> Tenants -> {tenant_name}

File: website/docs/d/tenant_to_cloud_account.html.markdown [hunk 32]
```diff
@@ -0,0 +1,44 @@
...
+data "aci_tenant_to_cloud_account" "example" {
+  tenant_dn  = aci_tenant.example.id
+  cloud_account_dn = aci_cloud_account_example.id
+}
+```
+
+## Argument Reference ##
+
+* `tenant_dn` - (Required) Distinguished name of the parent Tenant object.
+* `cloud_account_dn` - (Optional) Target-dn. The distinguished name of the target Tenant's Cloud Account object.
```
Comment (lhercot) [inline website/docs/d/tenant_to_cloud_account.html.markdown:38]: (Optional) The distinguished name of the target Cloud Account object.

File: website/docs/r/cloud_account.html.markdown [hunk 33]
```diff
@@ -0,0 +1,65 @@
...
+Manages ACI Cloud Account
+
+## API Information ##
+
+* `Class` - cloudAccount
+* `Distinguished Name` - uni/tn-{tenant_name}/act-[{account_id}]-vendor-{vendor}
+
+## GUI Information ##
+
+* `Location` - Cloud APIC -> Application Management -> Tenants
```
Comment (lhercot) [inline website/docs/r/cloud_account.html.markdown:21]: Cloud APIC -> Application Management -> Tenants -> {tenant_name}

File: website/docs/d/tenant_to_cloud_account.html.markdown [hunk 34]
```diff
@@ -0,0 +1,44 @@
...
+* `Location` - Cloud APIC -> Application Management -> Tenants 
+
+
+
+## Example Usage ##
+
+```hcl
+data "aci_tenant_to_cloud_account" "example" {
+  tenant_dn  = aci_tenant.example.id
+  cloud_account_dn = aci_cloud_account_example.id
```
Comment (lhercot) [inline website/docs/d/tenant_to_cloud_account.html.markdown:31]: aci_cloud_account.example.id

File: website/docs/r/cloud_account.html.markdown [hunk 35]
```diff
@@ -0,0 +1,65 @@
...
+
+
+## Example Usage ##
+
+```hcl
+resource "aci_cloud_account" "example" {
+  tenant_dn  = aci_tenant.example.id
+  account_id  = "example"
+  vendor  = "example"
```
Comment (lhercot) [inline website/docs/r/cloud_account.html.markdown:30]: example is not a correct vendor

File: website/docs/r/cloud_account.html.markdown [hunk 36]
```diff
@@ -0,0 +1,65 @@
...
+  access_type = "managed"
+  account_id = "example"
```
Comment (lhercot) [inline website/docs/r/cloud_account.html.markdown:32]: why is account_id and vendor twice?

File: website/docs/r/cloud_account.html.markdown [hunk 37]
```diff
@@ -0,0 +1,65 @@
...
+  vendor = "azure"
+  cloud_rs_account_to_access_policy = aci_resource.example.id
```
Comment (lhercot) [inline website/docs/r/cloud_account.html.markdown:34]: put one of the dn of the access policy as we don't have a resource for it.

File: website/docs/r/cloud_account.html.markdown [hunk 38]
```diff
@@ -0,0 +1,65 @@
...
+  cloud_rs_credentials = aci_resource.example.id
```
Comment (lhercot) [inline website/docs/r/cloud_account.html.markdown:35]: aci_cloud_credentials.example.id

File: website/docs/r/cloud_account.html.markdown [hunk 39]
```diff
@@ -0,0 +1,65 @@
...
+}
+```
+
+## Argument Reference ##
+
+* `tenant_dn` - (Required) Distinguished name of the parent Tenant object.
+* `account_id` - (Required) ID of the Cloud Account object.
+* `vendor` - (Required) Vendor of the Cloud Account object.
```
Comment (lhercot) [inline website/docs/r/cloud_account.html.markdown:43]: is there allowed values for vendor?

Comment (lhercot) [reply website/docs/r/cloud_account.html.markdown:43]: Copy text from optional vendor attribute

File: website/docs/r/cloud_account.html.markdown [hunk 40]
```diff
@@ -0,0 +1,65 @@
...
+* `name` - (Optional) Name of the Cloud Account object.
+* `annotation` - (Optional) Annotation of the Cloud Account object.
+* `access_type` - (Optional) Authentication to the Cloud Account (managed=no credentials required (IAM), credentials=using accessKeys). Allowed values are "credentials", "managed". Default value is "managed". Type: String.
+* `account_id` - (Optional) Account id. An object identifier.
+* `vendor` - (Optional) The Cloud Account vendor of the controller. Allowed values are "aws", "azure", "gcp", "unknown", and default value is "aws". Type: String.
```
Comment (lhercot) [inline website/docs/r/cloud_account.html.markdown:48]: why twice?

File: website/docs/r/cloud_account.html.markdown [hunk 41]
```diff
@@ -0,0 +1,65 @@
...
+```
+
+## Argument Reference ##
+
+* `tenant_dn` - (Required) Distinguished name of the parent Tenant object.
+* `account_id` - (Required) ID of the Cloud Account object.
+* `vendor` - (Required) Vendor of the Cloud Account object.
+* `name` - (Optional) Name of the Cloud Account object.
+* `annotation` - (Optional) Annotation of the Cloud Account object.
+* `access_type` - (Optional) Authentication to the Cloud Account (managed=no credentials required (IAM), credentials=using accessKeys). Allowed values are "credentials", "managed". Default value is "managed". Type: String.
```
Comment (lhercot) [inline website/docs/r/cloud_account.html.markdown:46]: Authentication type for the Cloud Account

File: website/docs/r/tenant_to_cloud_account.html.markdown [hunk 42]
```diff
@@ -0,0 +1,49 @@
...
+Manages ACI Tenant to Cloud Account association
+
+## API Information ##
+
+* `Class` - fvRsCloudAccount
+* `Distinguished Name` - uni/tn-{tenant_name}/rsCloudAccount
+
+## GUI Information ##
+
+* `Location` - Cloud APIC -> Application Management -> Tenants
```
Comment (lhercot) [inline website/docs/r/tenant_to_cloud_account.html.markdown:21]: Cloud APIC -> Application Management -> Tenants -> {tenant_name}

File: website/docs/r/tenant_to_cloud_account.html.markdown [hunk 43]
```diff
@@ -0,0 +1,49 @@
...
+
+* `Location` - Cloud APIC -> Application Management -> Tenants 
+
+
+## Example Usage ##
+
+```hcl
+resource "aci_tenant_to_cloud_account" "example" {
+  tenant_dn  = aci_tenant.example.id
+  cloud_account_dn = aci_cloud_account_example.id
```
Comment (lhercot) [inline website/docs/r/tenant_to_cloud_account.html.markdown:29]: aci_cloud_account.example.id

File: website/docs/r/tenant_to_cloud_account.html.markdown [hunk 44]
```diff
@@ -0,0 +1,49 @@
...
+}
+```
+
+## Argument Reference ##
+
+* `tenant_dn` - (Required) Distinguished name of the parent Tenant object.
+* `cloud_account_dn` - (Optional) Target-dn. The distinguished name of the target Tenant's Cloud Account object.
```
Comment (lhercot) [inline website/docs/r/tenant_to_cloud_account.html.markdown:36]: (Optional) The distinguished name of the target Cloud Account object.

**Review by lhercot — CHANGES_REQUESTED** [review]

File: website/docs/d/cloud_credentials.html.markdown [hunk 45]
```diff
@@ -0,0 +1,49 @@
...
+page_title: "ACI: aci_cloud_credentials"
+sidebar_current: "docs-aci-data-source-aci_cloud_credentials"
+description: |-
+  Data source for Cloud ACI Access Credential to manage the cloud resources
+---
+
+# aci_cloud_credentials #
+
+Data source for Cloud ACI Access Credential to manage the cloud resources
+Note: This resource is supported in Cloud APIC only.
```
Comment (lhercot) [inline website/docs/d/cloud_credentials.html.markdown:13]: We probably want this note in the other ones too.
It is a datasource here though

File: website/docs/d/cloud_credentials.html.markdown [hunk 46]
```diff
@@ -0,0 +1,49 @@
...
+data "aci_cloud_credentials" "example" {
+  tenant_dn  = aci_tenant.example.id
+  name  = "example"
+}
+```
+
+## Argument Reference ##
+
+* `tenant_dn` - (Required) Distinguished name of the parent Tenant object.
+* `name` - (Required) Name of the Access Credential object used to manage the cloud resources.
```
Comment (lhercot) [inline website/docs/d/cloud_credentials.html.markdown:38]: Name of the Cloud Credentials object used...

File: website/docs/d/cloud_credentials.html.markdown [hunk 47]
```diff
@@ -0,0 +1,49 @@
...
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the Access Credential to manage the cloud resources.
```
Comment (lhercot) [inline website/docs/d/cloud_credentials.html.markdown:41]: of the Cloud Credentials object.

File: website/docs/d/cloud_credentials.html.markdown [hunk 48]
```diff
@@ -0,0 +1,49 @@
...
+* `annotation` - (Optional) Annotation of the Access Credential object to manage the cloud resources.
```
Comment (lhercot) [inline website/docs/d/cloud_credentials.html.markdown:42]: of the Cloud Credentials object.

File: website/docs/d/cloud_credentials.html.markdown [hunk 49]
```diff
@@ -0,0 +1,49 @@
...
+* `name_alias` - (Optional) Name Alias of the Access Credential object to manage the cloud resources.
```
Comment (lhercot) [inline website/docs/d/cloud_credentials.html.markdown:43]: of the Cloud Credentials object.

File: website/docs/d/cloud_credentials.html.markdown [hunk 50]
```diff
@@ -0,0 +1,49 @@
...
+* `client_id` - (Optional) Client ID of the Access Credential object.
```
Comment (lhercot) [inline website/docs/d/cloud_credentials.html.markdown:44]: of the Cloud Credentials object.

File: website/docs/d/cloud_credentials.html.markdown [hunk 51]
```diff
@@ -0,0 +1,49 @@
...
+* `email` - (Optional) email address of the locally-authenticated user.
+* `http_proxy` - (Optional) Http Proxy to connect to cloud provider.
```
Comment (lhercot) [inline website/docs/d/cloud_credentials.html.markdown:46]: used to connect to the cloud provider

File: website/docs/r/cloud_account.html.markdown [hunk 52]
```diff
@@ -0,0 +1,61 @@
...
+  cloud_rs_credentials = aci_cloud_credentials.example.id
+}
+```
+
+## Argument Reference ##
+
+* `tenant_dn` - (Required) Distinguished name of the parent Tenant object.
+* `name` - (Optional) Name of the Cloud Account object.
+* `annotation` - (Optional) Annotation of the Cloud Account object.
+* `access_type` - (Optional) Authentication type for the Cloud Account (managed=no credentials required (IAM), credentials=using accessKeys). Allowed values are "credentials", "managed". Type: String.
```
Comment (lhercot) [inline website/docs/r/cloud_account.html.markdown:42]: are "credentials" and "managed"

File: website/docs/r/cloud_account.html.markdown [hunk 53]
```diff
@@ -0,0 +1,61 @@
...
+* `account_id` - (Required) ID of the Cloud Account object.
+* `vendor` - (Required) Vendor of the Cloud Account object. Allowed values are "aws", "azure", "gcp", "unknown". Type: String.
```
Comment (lhercot) [inline website/docs/r/cloud_account.html.markdown:44]: and "unknown".

File: website/docs/r/cloud_account.html.markdown [hunk 54]
```diff
@@ -0,0 +1,61 @@
...
+
+* `relation_cloud_rs_account_to_access_policy` - (Optional) Represents the relation to a Relation to the Access policy to be used (class cloudAccessPolicy). Relation to CloudAccessPolicy cardianity is n-1 Type: String.
```
Comment (lhercot) [inline website/docs/r/cloud_account.html.markdown:46]: Represents the relation to the Access policy to be used (class cloudAccessPolicy). Type: String.

File: website/docs/r/cloud_account.html.markdown [hunk 55]
```diff
@@ -0,0 +1,61 @@
...
+
+
+* `relation_cloud_rs_credentials` - (Optional) Represents the relation to a Credentials to use to manage the account (class cloudCredentials). If access type is credentials, relation to the credentials to use Type: String.
```
Comment (lhercot) [inline website/docs/r/cloud_account.html.markdown:49]: to Cloud Credentials to use ...
If the access_type is  credentials, this relation attribute is required. Type: String.

File: website/docs/r/cloudad.html.markdown [hunk 56]
```diff
@@ -0,0 +1,50 @@
+---
+subcategory: "Cloud"
+layout: "aci"
+page_title: "ACI: aci_cloud_ad"
+sidebar_current: "docs-aci-resource-cloud_ad"
+description: |-
+  Manages Cloud ACI Active Directory
```
Comment (lhercot) [inline website/docs/r/cloudad.html.markdown:7]: ACI Cloud Active Directory

File: website/docs/r/cloudad.html.markdown [hunk 57]
```diff
@@ -0,0 +1,50 @@
...
+---
+
+# aci_cloud_ad #
+
+Manages Cloud ACI Active Directory
```
Comment (lhercot) [inline website/docs/r/cloudad.html.markdown:12]: ACI Cloud Active Directory

File: website/docs/r/cloudad.html.markdown [hunk 58]
```diff
@@ -0,0 +1,50 @@
...
+* `Location` - Cloud APIC -> Application Management -> Tenants  -> {tenant_name}
+
+
+## Example Usage ##
+
+```hcl
+resource "aci_cloud_ad" "example" {
+  tenant_dn  = aci_tenant.example.id
+  active_directory_id  = "example"
+
```
Comment (lhercot) [inline website/docs/r/cloudad.html.markdown:30]: remove empty line

File: website/docs/r/cloudad.html.markdown [hunk 59]
```diff
@@ -0,0 +1,50 @@
...
+}
+```
+
+## Argument Reference ##
+
+* `tenant_dn` - (Required) Distinguished name of the parent Tenant object.
+* `active_directory_id` - (Required) Active_directory_id of the object Active Directory.
```
Comment (lhercot) [inline website/docs/r/cloudad.html.markdown:37]: (Required) Id of the Azure Active Directory.

File: website/docs/r/cloudcredentials.html.markdown [hunk 60]
```diff
@@ -0,0 +1,64 @@
...
+  rsa_private_key = "rsa_key"
+  cloud_rs_ad = aci_resource.example.id
+}
+```
+
+## Argument Reference ##
+
+* `tenant_dn` - (Required) Distinguished name of the parent Tenant object.
+* `name` - (Required) Name of the Access Credential object to manage the cloud resources.
+* `annotation` - (Optional) Annotation of the Access Credential object to manage the cloud resources.
```
Comment (lhercot) [inline website/docs/r/cloudcredentials.html.markdown:44]: of the Cloud Credentials object used to manage the cloud resources.

File: website/docs/r/cloudcredentials.html.markdown [hunk 61]
```diff
@@ -0,0 +1,64 @@
...
+  key_id = cloud_key_id"
+  rsa_private_key = "rsa_key"
+  cloud_rs_ad = aci_resource.example.id
+}
+```
+
+## Argument Reference ##
+
+* `tenant_dn` - (Required) Distinguished name of the parent Tenant object.
+* `name` - (Required) Name of the Access Credential object to manage the cloud resources.
```
Comment (lhercot) [inline website/docs/r/cloudcredentials.html.markdown:43]: of the Cloud Credentials object used to manage the cloud resources.

File: website/docs/r/cloudcredentials.html.markdown [hunk 62]
```diff
@@ -0,0 +1,64 @@
...
+* `annotation` - (Optional) Annotation of the Access Credential object to manage the cloud resources.
+* `client_id` - (Optional) Client ID of the Access Credential object.
+* `email` - (Optional) email address of the locally-authenticated user.
+* `http_proxy` - (Optional) Http Proxy to connect to cloud provider.
```
Comment (lhercot) [inline website/docs/r/cloudcredentials.html.markdown:47]: to the cloud provider

**Review by lhercot — CHANGES_REQUESTED** [review]

File: aci/resource_aci_cloudcredentials.go [hunk 63]
```diff
@@ -0,0 +1,333 @@
...
+	err := aciClient.Save(cloudCredentials)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	checkDns := make([]string, 0, 1)
+
+	if relationTocloudRsAD, ok := d.GetOk("relation_cloud_rs_ad"); ok {
+		relationParam := relationTocloudRsAD.(string)
+		checkDns = append(checkDns, relationParam)
+
```
Comment (lhercot) [inline aci/resource_aci_cloudcredentials.go:183]: remove empty line

File: website/docs/d/cloud_credentials.html.markdown [hunk 64]
```diff
@@ -0,0 +1,49 @@
...
+* `tenant_dn` - (Required) Distinguished name of the parent Tenant object.
+* `name` - (Required) Name of the Cloud Credential object used to manage the cloud resources.
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the Cloud Credential to manage the cloud resources.
+* `annotation` - (Optional) Annotation of the Cloud Credential object to manage the cloud resources.
+* `name_alias` - (Optional) Name Alias of the Cloud Credential object to manage the cloud resources.
+* `client_id` - (Optional) Client ID of the Cloud Credential object.
+* `email` - (Optional) email address of the locally-authenticated user.
+* `http_proxy` - (Optional) Http Proxy to connect to the cloud provider.
```
Comment (lhercot) [inline website/docs/d/cloud_credentials.html.markdown:46]: HTTP Proxy

File: website/docs/r/cloudcredentials.html.markdown [hunk 65]
```diff
@@ -0,0 +1,65 @@
...
+
+* `tenant_dn` - (Required) Distinguished name of the parent Tenant object.
+* `name` - (Required) Name of the Cloud Credential object to manage the cloud resources.
+* `annotation` - (Optional) Annotation of the Cloud Credential object to manage the cloud resources.
+* `client_id` - (Optional) Client ID of the Cloud Credential object.
+* `email` - (Optional) email address of the locally-authenticated user.
+* `http_proxy` - (Optional) Http Proxy to connect to the cloud provider. 
+* `key` - (Optional) The Secret key or password used to uniquely identify the cloud resource configuration object.
+* `key_id` - (Optional) The Access key ID of the cloud resource.
+* `rsa_private_key` - (Optional)  RSA Secret Key of the cloud resource. Allowed values are and default value is "".
```
Comment (lhercot) [inline website/docs/r/cloudcredentials.html.markdown:51]: I think the allowed value part can be removed as it is empty.

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by lhercot — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM
